workers/unity/Assets/Gdk/Core/Tests/Editmode/Components/OptionTests.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using NUnit.Framework;
+using Unity.Mathematics;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class OptionTests
+    {
+        [Test]
+        public void Parameterless_constructor_creates_empty_option()
+        {
+            var option = new Option<bool>();
+            Assert.AreEqual((bool1)false, option.HasValue);
+        }
+
+        [Test]
+        public void Constructor_with_parameters_creates_option_with_payload()
+        {
+            var payload = true;
+            var option = new Option<bool>(payload);
+            Assert.AreEqual((bool1)true, option.HasValue);
+            Assert.AreEqual(payload, option.Value);
+        }
+
+        [Test]
+        public void Value_can_be_read()
+        {
+            var payload = true;
+            var option = new Option<bool>(payload);
+            Assert.AreEqual(payload, option.Value);
+        }
+
+        [Test]
+        public void Value_can_be_set()
+        {
+            var payload = true;
+            var option = new Option<bool>();
+            option.Value = payload;
+            Assert.AreEqual((bool1)true, option.HasValue);
+            Assert.AreEqual(payload, option.Value);
+        }
+
+        [Test]
+        public void Accessing_value_of_empty_option_throws()
+        {
+            var option = new Option<bool>();
+            Assert.AreEqual((bool1)false, option.HasValue);
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                var value = option.Value;
+            });
+        }
+
+        [Test]
+        public void Setting_null_value_throws()
+        {
+            var option = new Option<string>();
+            Assert.AreEqual((bool1)false, option.HasValue);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                option.Value = null;
+            });
+        }
+
+        [Test]
+        public void Option_payload_can_be_cleared()
+        {
+            var option = new Option<bool>(true);
+            Assert.AreEqual((bool1)true, option.HasValue);
+            option.Clear();
+            Assert.AreEqual((bool1)false, option.HasValue);
+        }
+
+        [Test]
+        public void TryGetValue_on_empty_option()
```

Comment:
Please rename this test method to give a hint on what it asserts, e.g. `TryGetValue_returns_false_when_empty`

Reply:
Also for the methods below too

Reply:
Fixed.